src/selectors/contents.js

```diff
@@ -64,21 +82,28 @@ export const getDirectories = createSelector([getContentsPerDirectories], conten
   ),
 )
 
-export const getFavoritesContents = createSelector([getAllContents], contents =>
-  contents.filter(c => c.isFavorite),
+export const getAugmentedFavoriteContents = createSelector(
+  [getAllContents, getFavorites],
+  (contents, favorites) =>
+    getAugmentedContents(contents.filter(content => favorites[content.id]), favorites),
 )
 
-export const getHistoryContents = createSelector([getAllContents], contents =>
-  contents
-    .filter(content => content.lastPlayed)
-    .sort((c1, c2) => (c1.lastPlayed > c2.lastPlayed ? -1 : 1)),
+export const getAugmentedHistoryContents = createSelector(
+  [getAllContents, getFavorites, getHistory],
+  (contents, favorites, history) =>
+    getAugmentedContents(
+      contents
+        .filter(content => history[content.id])
```

Comment:
Since we're here for performance reasons:

```javascript
filter(content => typeof history[content.id] !== "undefined")
```

Otherwise, JavaScript will test againt null, undefined, void, empty string & so on.
And testing against `undefined` directly will trigger some checks also since `undefined` could be redefined.

Reply:
Hmmm why not, but I'd have to add that reasoning in the comments to understand why this check would be written like this… Is it that bad, perf-wise ?